CMakeLists.txt

```diff
@@ -53,69 +56,68 @@ add_library(component
   src/ftxui/component/menu.cpp
   src/ftxui/component/radiobox.cpp
   src/ftxui/component/screen_interactive.cpp
-  src/ftxui/component/toggle.cpp
-)
+  src/ftxui/component/toggle.cpp)
 
 target_link_libraries(dom PUBLIC screen)
 target_link_libraries(component PUBLIC dom)
 target_link_libraries(component PUBLIC Threads::Threads)
 
 foreach(lib screen dom component)
-  target_include_directories(${lib}
-    PUBLIC
-      $<INSTALL_INTERFACE:include>    
-      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-    PRIVATE src
-  )
+  target_include_directories(
+    ${lib}
+    PUBLIC $<INSTALL_INTERFACE:include>
+           $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    PRIVATE src)
   target_compile_features(${lib} PUBLIC cxx_std_17)
   target_compile_options(${lib} PRIVATE -Wall -Werror -Wextra)
 endforeach()
 
-include(GNUInstallDirs)
-install(TARGETS screen dom component
-    EXPORT ftxui-export
+if(FTXUI_ENABLE_INSTALL)
+  include(GNUInstallDirs)
+  install(
+    TARGETS screen dom
+    component EXPORT
+    ftxui-export
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/ftxui/
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}/ftxui/
-)
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}/ftxui/)
 
-install(DIRECTORY include/ftxui DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+  install(DIRECTORY include/ftxui DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
-include(CMakePackageConfigHelpers)
-write_basic_package_version_file(
-  ftxui-config.cmake
-  VERSION ${PACKAGE_VERSION}
-  COMPATIBILITY AnyNewerVersion
-)
+  include(CMakePackageConfigHelpers)
+  write_basic_package_version_file(
+    ftxui-config.cmake
+    VERSION ${PACKAGE_VERSION}
+    COMPATIBILITY AnyNewerVersion)
 
-install(EXPORT ftxui-export
-  FILE ftxui-config.cmake
-	NAMESPACE ftxui::
-	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/ftxui
-)
+  install(
+    EXPORT ftxui-export
+    FILE ftxui-config.cmake
+    NAMESPACE ftxui::
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/ftxui)
+endif()
 
 # Note: For gtest, please follow:
-# https://stackoverflow.com/questions/24295876/cmake-cannot-find-a-googletest-required-library
-find_package(GTest)
-if (GTEST_FOUND AND THREADS_FOUND)
-  add_executable(dom_tests
-    tests/gauge_test.cpp
-    tests/hbox_test.cpp
-    tests/text_test.cpp
-    tests/vbox_test.cpp
-  )
-
-  target_link_libraries(dom_tests
-    PRIVATE dom
-    PRIVATE Threads::Threads
-    PRIVATE ${GTEST_BOTH_LIBRARIES}
-  )
-
-  target_include_directories(dom_tests
-    PRIVATE ${GTest_INCLUDE_DIRS}
-  )
-  gtest_discover_tests(dom_tests)
-  add_test(dom_tests dom_tests)
-
+# https://stackoverflow.com/questions/24295876/cmake-cannot-find-a-googletest-
+# required-library
+if(FTXUI_BUILD_TESTS)
```

Comment:
Nit: I would move this inside the "if" line 104. This will remove one indentation level.